scalanet/ut/src/io/iohk/scalanet/peergroup/DynamicTLSPeerGroupSpec.scala

```diff
@@ -314,6 +312,51 @@ class DynamicTLSPeerGroupSpec extends AsyncFlatSpec with BeforeAndAfterAll {
         }
     }
   }
+  it should "handle messages with prepended messages id-s" in taskTestCase {
+    implicit val s = Codec[TestMessage[String]]
+    (for {
+      client <- DynamicTLSPeerGroup[TestMessage[String]](getCorrectConfig())
+      server <- DynamicTLSPeerGroup[TestMessage[String]](getCorrectConfig())
+      ch1 <- client.client(server.processAddress)
+    } yield (client, server, ch1)).use {
+      case (client, server, ch1) =>
+        for {
+          result <- Task.parZip2(
+            ch1.sendMessage(TestMessage.Foo("hello")),
+            server.serverEventObservable.collectChannelCreated.mergeMap {
+              case (channel, release) =>
+                channel.channelEventObservable.guarantee(release)
+            }.headL
+          )
+          (_, eventReceived) = result
+        } yield {
+          assert(eventReceived == MessageReceived(TestMessage.Foo("hello")))
+        }
+    }
+  }
+
+  it should "inform user about too large frame" in taskTestCase {
+    (for {
+      client <- DynamicTLSPeerGroup[Byte](getCorrectConfig())
+      // 1byte message + 4 bytes length field gives frame of size 5, so maxFrameLength = 4 should end with error
+      server <- DynamicTLSPeerGroup[Byte](getCorrectConfig(maxFrameLength = 4))
+      ch1 <- client.client(server.processAddress)
+    } yield (client, server, ch1)).use {
+      case (client, server, ch1) =>
+        for {
+          result <- Task.parZip2(
+            ch1.sendMessage(1.toByte),
+            server.serverEventObservable.collectChannelCreated.mergeMap {
+              case (channel, release) =>
+                channel.channelEventObservable.guarantee(release)
+            }.headL
```

Comment:
Maybe this pattern could be added as an extension method somewhere in the test, it appears so often.